core/jvm/src/main/scala/fs2/compress.scala

```diff
@@ -75,197 +105,658 @@ object compress {
+        case inflatedBytes if inflatedBytes <= -2 =>
+          Pull.output(chunk)
+        case inflatedBytes if inflatedBytes == -1 =>
+          Pull.done
+        case inflatedBytes if inflatedBytes < inflatedBuffer.length =>
+          if (inflater.finished()) {
+            inflater.getRemaining match {
+              case bytesRemaining if bytesRemaining > 0 =>
+                Pull.output(asChunkBytes(inflatedBuffer, inflatedBytes)) >>
+                  Pull.output(
+                    Chunk.Bytes(
+                      bytesChunk.values,
+                      bytesChunk.offset + bytesChunk.length - bytesRemaining,
+                      bytesRemaining
+                    )
+                  ) >> Pull.done
+              case _ =>
+                Pull.output(asChunkBytes(inflatedBuffer, inflatedBytes)) >> Pull.done
+            }
+          } else Pull.output(asChunkBytes(inflatedBuffer, inflatedBytes))
+        case inflatedBytes =>
+          Pull.output(asChunkBytes(inflatedBuffer, inflatedBytes)) >> pull()
+      }
+    }
+
+    pull()
+  }
+
+  private def _inflate_stream[F[_]](
+      inflater: Inflater,
+      inflatedBufferSize: Int,
+      crc32: Option[CRC32]
+  )(implicit ev: RaiseThrowable[F]): Stream[F, Byte] => Pull[F, Byte, Unit] =
+    _.pull.unconsNonEmpty.flatMap {
+      case Some((deflatedChunk, deflatedStream)) =>
+        _inflate_chunk(deflatedChunk, inflater, inflatedBufferSize, crc32) >> _inflate_stream(
+          inflater,
+          inflatedBufferSize,
+          crc32
+        )(ev)(deflatedStream)
       case None =>
         if (!inflater.finished)
           Pull.raiseError[F](new DataFormatException("Insufficient data"))
         else {
-          inflater.end(); Pull.done
+          Pull.done
         }
     }
 
-  @tailrec
-  private def _inflate_collect(
-      inflater: Inflater,
-      buffer: Array[Byte],
-      acc: ArrayBuffer[Byte]
-  ): ArrayBuffer[Byte] =
-    if (inflater.finished || inflater.needsInput) acc
-    else {
-      val count = inflater.inflate(buffer)
-      _inflate_collect(inflater, buffer, acc ++ buffer.iterator.take(count))
-    }
-
   /**
-    * Returns a pipe that incrementally compresses input into the GZIP format
-    * by delegating to `java.util.zip.GZIPOutputStream`. Output is compatible
-    * with the GNU utils `gunzip` utility, as well as really anything else that
-    * understands GZIP. Note, however, that the GZIP format is not "stable" in
-    * the sense that all compressors will produce identical output given
-    * identical input. Part of the header seeding is arbitrary and chosen by
-    * the compression implementation. For this reason, the exact bytes produced
-    * by this pipe will differ in insignificant ways from the exact bytes produced
-    * by a tool like the GNU utils `gzip`.
+    * Returns a `Pipe` that gzips (compresses) its input elements as defined by RFC 1952:
+    * https://www.ietf.org/rfc/rfc1952.txt
```

Comment:
This is actually clearer than what was there before? Shouldn't we just add the RFC link to what's there?

Reply:
Original documentation restored with the addition of the RFC link.